_posts/2018-08-03-bashing-through-minikube.markdown

```diff
@@ -0,0 +1,87 @@
+3. `local-run.sh`
+    This script deploys Kubernetes manifests to Minikube, either via Helm, Kubectl or some other tooling. Also you can run integration tests here.
+4. `local-nuke.sh`
+    This script remove the resources from Kubernetes and deletes the Docker image artifact
+5. `publish.sh`
+    This script takes your artifacts and publishes them to your artifact repository.
+
+This leads into the core of your continuous integration configuration being `build.sh && publish.sh`
+
+### 2. Keep your Containers Small
+Using multi-stage Docker builds, i.e one container to build and one minimal container for runtime, will keep your feedback cycles fast. For example, with java your `build` container can have the JDK installed with all your testing dependencies, but then you can create a JRE only Alpine container that can be about 5-10x smaller.
+
+Watch this [video](https://www.youtube.com/watch?v=wGz_cbtCiEA) for more on why small containers are best practice.
+
+### 3. Have your Container Entrypoint and Command be targeted
+The containers that you are expecting to run in your cluster should be single purpose by design, that means that you can have your entrypoint and command set so you don't have to manage that outside the Dockerfile itself.
+
+### 4. Exposing Ports, e.g. I want to see my Database
+
+`kubectl port-forward $(kubectl get pods | grep db | awk {'print $1'}) 5432:5432`
+
+### 5. Alias away common command chains
+Minikube, alas, is not perfect which means you have to restart it from scratch on occasion. Also when iterating on the scripts from pattern #1 it's helpful to make sure they work on a fresh machine.
+
+`alias reset-minikube='minikube delete && minikube start && helm init`
+
+Also for iterating on your integration test feedback loop:
+
+`alias warm-reload='./local-nuke.sh && ./local-build.sh && ./local-run.sh'`
+
+And for managing the docker environments:
+
+`alias local-docker='unset DOCKER_API_VERSION DOCKER_TLS_VERIFY DOCKER_CERT_PATH DOCKER_API_VERSION DOCKER_HOST'`
+
+
+
+## Anti-Patterns
+### 1. Every project in a repository should be testable in isolation
+With Docker it is so trivial to spin up a database container that people can make the mistake of making all their component tests require an actual database to function. This complicates matters significantly especially when you add parallel docker environments between minikube and your local machine.
+
+### 2. Don't rely on on being able to run your entire stack locally
+When you have a small number of services it is possible to run them all locally at once, however this stops being the case quickly, and is more difficult to retroactively fix then to stop this at the start.
+
+The need to do this is generally driven by insufficient testing, or testing at the wrong isolation levels. For testing across repositories leave that to Continuous Delivery.
+
+
+## Wrap Up
```

Comment:
your previous headings throughout the article are `Sentence case`. This heading is `Title Case`. Up to you which you want to use, but you should pick one and make it consistent through the article.